js/main.js

```diff
@@ -0,0 +1,132 @@
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    generateDomTree(xhr);
+    var xhrLoaded = xhr;
+    return xhrLoaded;
+  });
+  xhr.send();
+}
+
+function generateDomTree(foodValues) {
+  $itemListDisplay.innerHTML = null;
+  if (foodValues.response.hints.length === 0) {
+    var announcement = document.createElement('h1');
+    var announcementText = document.createTextNode('There are 0 results!');
+    announcement.append(announcementText);
+    $itemListDisplay.append(announcement);
+    return;
+  }
+  for (var i = 0; i < foodValues.response.hints.length; i++) {
+    var foodLabel = foodValues.response.hints[i].food.label;
+    var foodImageSrc = foodValues.response.hints[i].food.image;
+    var nutrients = foodValues.response.hints[i].food.nutrients;
+
+    var macroList = {};
+
+    if (foodValues.response.hints[i].food.image === undefined) {
+      foodImageSrc = 'https://vetmed.tamu.edu/news/wp-content/uploads/sites/9/2020/09/Llama-and-Alpaca-Pet-Talk.jpg';
+    }
+
+    var calories = Math.round(nutrients.ENERC_KCAL, 2);
+    var carbs = Math.round(nutrients.CHOCDF, 2);
+    var fats = Math.round(nutrients.FAT, 2);
+    var protein = Math.round(nutrients.PROCNT, 2);
+
+    macroList.calories = calories;
+    macroList.carbs = carbs;
+    macroList.fats = fats;
+    macroList.protein = protein;
+
+    var individualItem = document.createElement('div');
+    var image = document.createElement('img');
+    var itemInfo = document.createElement('div');
+    var itemName = document.createElement('h3');
+    var options = document.createElement('div');
+    var favoriteIcon = document.createElement('i');
+    var itemDivider = document.createElement('hr');
+
+    individualItem.setAttribute('class', 'row-90 item');
+    image.setAttribute('class', 'column item-img');
+    image.setAttribute('src', foodImageSrc);
+    image.setAttribute('alt', 'Item Preview');
+    itemInfo.setAttribute('class', 'column item-desc-container');
+    itemName.setAttribute('class', 'item-name ellipsis');
+    options.setAttribute('class', 'item-options');
+    favoriteIcon.setAttribute('class', 'far fa-heart');
+    itemDivider.setAttribute('class', 'item-divider');
+
+    individualItem.append(image);
+
+    var itemNameText = document.createTextNode(foodLabel);
+    itemName.append(itemNameText);
+    itemInfo.append(itemName);
+    individualItem.append(itemInfo);
+
+    for (var macro in macroList) {
+      var itemDesc = document.createElement('h4');
+      itemDesc.setAttribute('class', 'item-desc');
+      var itemDescText = document.createTextNode(`${macro}: ` + macroList[macro]);
+      itemDesc.append(itemDescText);
+      itemInfo.append(itemDesc);
+      individualItem.append(itemInfo);
+    }
+
+    options.append(favoriteIcon);
+    individualItem.append(options);
+
+    $itemListDisplay.append(individualItem);
+    $itemListDisplay.append(itemDivider);
+  }
+
+}
```

Comment:
Whoops! You don't want to append directly to the DOM here, you want to `return` the completed DOM tree so you can `append` the result where the function is called. By just appending directly, this function can only ever be used to append to this specific element. Reusability is the goal!